storage/reads/array_cursor.go

```diff
@@ -22,10 +22,23 @@ func newAggregateArrayCursor(ctx context.Context, agg *datatypes.Aggregate, curs
 	}
 
 	switch agg.Type {
+	case datatypes.AggregateTypeCount:
+		return newCountArrayCursor(cursor)
+	default:
+		panic("invalid aggregate")
+	}
+}
+
+func newWindowAggregateArrayCursor(ctx context.Context, req *datatypes.ReadWindowAggregateRequest, cursor cursors.Cursor) cursors.Cursor {
+	if cursor == nil {
+		return nil
+	}
+
+	switch req.Aggregate[0].Type {
 	case datatypes.AggregateTypeSum:
 		return newSumArrayCursor(cursor)
 	case datatypes.AggregateTypeCount:
-		return newCountArrayCursor(cursor)
+		return newWindowCountArrayCursor(cursor, req)
 	default:
 		// TODO(sgc): should be validated higher up
 		panic("invalid aggregate")
```

Comment:
Is is worth creating an issue to validate this sooner? Could this introduce a bug?

Reply:
I don't _believe_ so. I think we saw that it _is_ validated higher up. This could probably just as well be a "unreachable" sort of panic.